include/nbl/asset/IAssetManager.h

```diff
@@ -203,28 +203,37 @@ class IAssetManager : public core::IReferenceCounted, public core::QuitSignallin
 		*/
         SAssetBundle getAssetInHierarchy(io::IReadFile* _file, const std::string& _supposedFilename, const IAssetLoader::SAssetLoadParams& _params, uint32_t _hierarchyLevel, IAssetLoader::IAssetLoaderOverride* _override)
         {
+            const uint32_t restoreLevels = (_hierarchyLevel >= _params.restoreLevels) ? 0u : (_params.restoreLevels - _hierarchyLevel);
+
             IAssetLoader::SAssetLoadParams params(_params);
             if (params.meshManipulatorOverride == nullptr)
             {
                 params.meshManipulatorOverride = m_meshManipulator.get();
             }
+            if (restoreLevels)
+            {
+                using flags_t = std::underlying_type_t<IAssetLoader::E_CACHING_FLAGS>;
+                flags_t mask = ~static_cast<flags_t>(0);
+                mask = core::bitfieldInsert<flags_t>(mask, IAssetLoader::ECF_DONT_CACHE_REFERENCES, 2u*_hierarchyLevel, 2u*restoreLevels);
+                params.cacheFlags = static_cast<IAssetLoader::E_CACHING_FLAGS>(params.cacheFlags & mask);
+            }
```

Comment:
you're only supposed to do this during the second call to `getAssetInHierarchy` meant for a reload (not the first call, because then you wont cache anything new you might load), remember?

DUPLICATE too, not DONT CACHE references

Reply:
oh wait wtf, you're masking out the DONT_CACHE bit on [hierarchyLevel,hierarchyLevel+restoreLevels) ?

Reply:
then for these levels
CACHE -> CACHE
DONT_CACHE -> CACHE ???
DUPLICATE -> CACHE ???

maan this bit arithmetic is confusing AF, I'd like to know its intended purpose

Reply:
masking is not *out*
ignoring DUPLICATE, leaving DONT_CACHE as is

Reply:
> then for these levels
> CACHE -> CACHE
> DONT_CACHE -> CACHE ???
> DUPLICATE -> CACHE ???
> 
> maan this bit arithmetic is confusing AF, I'd like to know its intended purpose

we've ben talking about this
CACHE -> CACHE
DONT_CACHE -> DONT_CACHE
DUPLICATE -> DONT_CACHE

Reply:
I still dont get why you feel the need to override DUPLICATE into DONT_CACHE of any level in the restore level range

If user/loader/override wanted a DUPLICATE, the better for US! Then at least we can load the relevant level assets in duplicate on first attempt, then they won't report being dummies (because freshly loaded duplicates never are) and we'll do the loading once instead of twice.

I see no downside to leaving the flags as they are on the preliminary load!

Reply:
Also by overriding DUPLICATE into DONT_CACHE you're violating what the user/loader/override wants, because they want a fresh asset, not a pointer to some old crap in the cache that you're now hellbent to restore.

Reply:
ok, you've convinced me